services/kubeapi.go

```diff
@@ -43,6 +43,10 @@ func upgradeKubeAPI(host hosts.Host, etcdHosts []hosts.Host, kubeAPIService v1.K
 	return err
 }
 
+func removeKubeAPI(host hosts.Host) error {
+	return docker.DoRemoveContainer(host.DClient, KubeAPIContainerName, host.AdvertisedHostname)
```

Comment:
Add logging to this and the similar ones ? Like you have with etcd ?

Reply:
Actually i do in https://github.com/rancher/rke/pull/23/files#diff-a1440968342b4da00653d6c74b21f48eR56 and in workerplane file